compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -625,16 +625,118 @@ impl<'a> FunctionContext<'a> {
         location: Location,
     ) -> ValueId {
         self.builder.set_location(location);
-
-        // To ensure that `value` is a valid `typ`, we insert an `Instruction::Truncate` instruction beforehand if
-        // we're narrowing the type size.
-        let incoming_type_size = self.builder.type_of_value(value).bit_size();
-        let target_type_size = typ.bit_size();
-        if target_type_size < incoming_type_size {
-            value = self.builder.insert_truncate(value, target_type_size, incoming_type_size);
-        }
-
-        self.builder.insert_cast(value, typ)
+        let incoming_type = self.builder.type_of_value(value);
+
+        let result = match (&incoming_type, typ) {
+            // Casting to field is safe
+            (_, NumericType::NativeField) => value,
```

Comment:
Can you add some tests for casting from signed types to `Field`? We can't really sign extend on these but we should have tests to enforce some behaviour here.

Reply:
Done